### PR TITLE
[WIP]Add CMP-9836 - Name field - naming conventions

### PR DESCRIPTION
**Pull request by jaryn**
{{pytest: cfme/tests/containers/test_naming_conventions.py -v --use-provider container}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 1b1041749457b5c61421d1ed6c522aca6fcb28aa:

`cfme/containers/provider/__init__.py`:
- :bangbang: Line 121:101: **E501** *line too long (107 > 100 characters)*

`cfme/tests/containers/test_naming_conventions.py`:
- :bangbang: Line 7:4: **E121** *continuation line under-indented for hanging indent*

`cfme/web_ui/__init__.py`:
- :bangbang: Line 2346:101: **E501** *line too long (101 > 100 characters)*

`utils/api.py`:
- :bangbang: Line 261:101: **E501** *line too long (107 > 100 characters)*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5fbdb5c737254fa1ef97e33b1e006beb716c53d5:

`cfme/tests/containers/test_naming_conventions.py`:
- :bangbang: Line 7:4: **E121** *continuation line under-indented for hanging indent*

`utils/api.py`:
- :bangbang: Line 252:68: **E111** *indentation is not a multiple of four*
- :bangbang: Line 252:68: **E113** *unexpected indentation*
- :bangbang: Line 263:68: **E111** *indentation is not a multiple of four*
- :bangbang: Line 263:68: **E113** *unexpected indentation*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c5191041ff91ceb0ecb518d4d994bf1ad82b68f3:

`utils/api.py`:
- :bangbang: Line 252:69: **E113** *unexpected indentation*
- :bangbang: Line 263:69: **E113** *unexpected indentation*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0659192ec84f09eb9d5c7d5cf7caaca4b5522cbf:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/common/provider.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from __future__ import unicode_literals
```
Comment (mfalesni) [inline cfme/common/provider.py]: You sure here?

**Comment by mfalesni** [comment]
@psav Doesn't this belong to the global scope of testing not just containers?

**Comment by mfalesni** [comment]
@jaryn please remove the changes to the rest module.

File: utils/api.py [hunk 2]
```diff
@@ -245,7 +248,8 @@ def find_by(self, **params):
     def _find_by_sqlfilter(self, **params):
         search_query = []
         for key, value in params.iteritems():
-            search_query.append("{} = {}".format(key, repr(str(value))))
+            # We may need to escape some chars here!
+            search_query.append("{} = '{}'".format(key, value))
```
Comment (mfalesni) [inline utils/api.py]: This is deprecated and is gone anyway, we shall switch to the externalized version of this library any moment. Remove tha changes on this file please then.

Comment (jaryn) [inline utils/api.py]: Without this change, the test fails when trying to find the flash message.

**Comment by dajoRH** [comment]
Lint report for commit ce80a8554c9d92eb4d6f22d9990970d436717378:

Everything seems all right :smile: .
*CFME QE Bot*

File: utils/api.py (hunk 2 above)
Comment (mfalesni) [inline utils/api.py]: But this file is going away and _find_by_sqlfilter shall not be invoked in the recent cfme versions.

File: cfme/containers/provider/__init__.py [hunk 3]
```diff
@@ -117,7 +118,8 @@ def __init__(self, name=None, credentials=None, key=None,
     def _on_detail_page(self):
         """ Returns ``True`` if on the providers detail page, ``False`` if not."""
         ensure_browser_open()
-        return sel.is_displayed('//div//h1[contains(., "{} (Summary)")]'.format(self.name))
+        return sel.is_displayed('//div//h1[contains(., "{} (Summary)")]'.format(
```
Comment (psav) [inline cfme/containers/provider/__init__.py]: Why altering this?

File: cfme/containers/provider/openshift.py [hunk 4]
```diff
@@ -40,7 +40,7 @@ def configloader(prov_config, prov_key):
         token_creds = OpenshiftProvider.process_credential_yaml_key(
             prov_config['credentials'], cred_type='token')
         return OpenshiftProvider(
-            name=prov_config['name'],
+            name=prov_config['name'].encode('utf-8'),
```
Comment (psav) [inline cfme/containers/provider/openshift.py]: If we do this in one place, we should do it throughout, I have oft wanted to create a generic config loader, but didn't yet get around to it

File: cfme/fixtures/pytest_selenium.py [hunk 5]
```diff
@@ -262,8 +262,8 @@ def element(o, **kwargs):
             if r_e is not None:
                 raise exceptions.CFMEExceptionOccured(
                     "Element {} not found on page because the following Rails error happened:\n{}"
-                    .format(str(o), r_e))
-        raise NoSuchElementException("Element {} not found on page.".format(str(o)))
+                    .format(o, r_e))
```
Comment (psav) [inline cfme/fixtures/pytest_selenium.py]: Why removing the explicit str cast here?

File: cfme/web_ui/__init__.py [hunk 6]
```diff
@@ -2342,7 +2342,8 @@ def a_cond(self):
             return "contains(normalize-space(@title), {name})"\
                 .format(name=quoteattr('Name: {}'.format(self._name)))
         else:
-            return "@title={name} or @data-original-title={name}".format(name=quoteattr(self._name))
+            return "@title={name} or @data-original-title={name}"\
```
Comment (psav) [inline cfme/web_ui/__init__.py]:
```python
            return "@title={name} or @data-original-title={name}".format(
                name=quoteattr(self._name))
```

**Comment by psav** [comment]
This PR seems to suggest it is adding/changing a test, but I see no test files being modified

File: cfme/containers/provider/openshift.py (hunk 4 above)
Comment (jaryn) [inline cfme/containers/provider/openshift.py]: Yeah, I was lazy there. We should convert all the loaded values to the utf-8. yaml has a way to do it

    import yaml
    def custom_str_constructor(loader, node):
        return loader.construct_scalar(node).encode('utf-8')
    yaml.add_constructor(u'tag:yaml.org,2002:str', custom_str_constructor)

but I didn't make it work together with the `yaycl`.

Maybe there is a way to walk trough the config tree after loading and convert all to utf, but I haven't found a way to do that yet.

File: cfme/fixtures/pytest_selenium.py (hunk 5 above)
Comment (jaryn) [inline cfme/fixtures/pytest_selenium.py]: Because it is unecessary.

File: cfme/web_ui/__init__.py (hunk 6 above)
Comment (jaryn) [inline cfme/web_ui/__init__.py]: I like what you are suggesting better, but I wanted to stay consistent with the above.

**Comment by dajoRH** [comment]
Lint report for commit be850686d17a9a9f4737a0fe7e5aa7495d4f585f:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by jaryn** [comment]
@psav Sorry. The file got lost somewhere. I re-added it.

**Comment by dajoRH** [comment]
Lint report for commit 4e3308cb75ffeba74a7df501887eff323f35a31f:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 978c15db22ee439e779f02f2ffe8c70ca556fe4d:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 048f088229c12ea06321cb4c34bf38f761347445:

Everything seems all right from lint perspective :smile: .

Commit inspection report:
**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9d98416c1adeff2ba1c277cefd70f581247278db:

`cfme/tests/containers/test_naming_conventions.py`:
- :bangbang: Line 20:1: **E302** *expected 2 blank lines, found 1*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 75c5c9a4ba0890da888847e5660c8f0759be3cdd:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4275a4d306aa20defa84a9219021015ab8cedc63:

`utils/appliance/__init__.py`:
- :red_circle: Line 30:1: **F401** *'trackerbot' imported but unused*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ed0df268a2ca4a6d322f302a261e4b9057efb7ab:

`utils/appliance/__init__.py`:
- :red_circle: Line 30:1: **F401** *'trackerbot' imported but unused*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0cb671856f6a8d4e6532f1a9dedc8cdea5df4dd8:

`utils/appliance/__init__.py`:
- :red_circle: Line 28:1: **F401** *'VM' imported but unused*
- :red_circle: Line 31:1: **F401** *'trackerbot' imported but unused*
- :red_circle: Line 38:1: **F401** *'get_crud' imported but unused*
- :red_circle: Line 38:1: **F401** *'get_mgmt' imported but unused*
- :red_circle: Line 50:5: **F401** *'setup_provider' imported but unused*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0cb671856f6a8d4e6532f1a9dedc8cdea5df4dd8:

`utils/appliance/__init__.py`:
- :red_circle: Line 28:1: **F401** *'VM' imported but unused*
- :red_circle: Line 31:1: **F401** *'trackerbot' imported but unused*
- :red_circle: Line 38:1: **F401** *'get_crud' imported but unused*
- :red_circle: Line 38:1: **F401** *'get_mgmt' imported but unused*
- :red_circle: Line 50:5: **F401** *'setup_provider' imported but unused*

Please, rectify these issues :smirk: .

Commit inspection report:
* 49ca18a862d63f4bc267ba424a1ebf3e73c301b1
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which can cause some problems during squashing. Please take **EXTRA care** when squashing!
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e516edafbc979cefe1929bc5b30980905359f978:

`cfme/tests/containers/test_naming_conventions.py`:
- :red_circle: Line 8:1: **F401** *'current_version' imported but unused*
- :bangbang: Line 12:1: **E122** *continuation line missing indentation or outdented*
- :bangbang: Line 13:1: **E122** *continuation line missing indentation or outdented*

`utils/providers.py`:
- :bangbang: Line 195:101: **E501** *line too long (108 > 100 characters)*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 1181bc0132995d32fcfb36e09332ce577b00f72a:

`cfme/tests/containers/test_naming_conventions.py`:
- :red_circle: Line 7:1: **F401** *'testgen' imported but unused*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 315005af5b30a86af557e454eab7ed9bfa973b03:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by jkrocil** [comment]
@mfalesni https://github.com/ManageIQ/manageiq-api-client-python/pull/9 merged, do we move this forward?

**Comment by mfalesni** [comment]
I rebased #3293 that should pull in the externalized client.

**Comment by dajoRH** [comment]
Lint report for commit 32c5132d32eb9b5a50b484e772dfe72f96c39610:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b7cf0fc965ebf44f333642e8f6f62390a7344ada:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by jkrocil** [comment]
Can we get this rebased, please? I'd like to move it forward.

**Comment by psav** [comment]
Can this be rebased please?

**Comment by jkrocil** [comment]
Closing for inactivity. Feel free to re-open once rebased.

**Pull request closed without merging**